### PR TITLE
Fixes #1159. Dialog must have a default button if none is provided.

### LINKED ISSUE #1159: Dialog must have a default button if none is provided.

### PR DESCRIPTION
**Pull request by BDisp**

### CODE REVIEW

**Comment by BDisp** [comment]
It was thinking on the MessageBox I did that. See in the All View Tester. This way you don't need to write the button parameter.

**Comment by BDisp** [comment]
I got the conclusion you are right because someone may not want to have buttons. I will revert this and I do not know if you want me to keep the typo fixes or you want me to close this PR.

**Comment by BDisp** [comment]
I ended up adding a functionality to specify which the default button for the MessageBox.

**Comment by BDisp** [comment]
Here is the change I've made:

default-button

**Comment by BDisp** [comment]
Only to remind that this PR won't break anything backward.

**Review by tig — APPROVED** [review]
> Only to remind that this PR won't break anything backward.

Yeah, I now agree. I was thinking that by adding another parameter between two existing parameters it might be a problem. But because both `int defaultButton = 0` and `params string [] buttons` are optional, and unambiguously typed, it's not an issue.

File: Terminal.Gui/Windows/MessageBox.cs [hunk 1]
```diff
@@ -88,10 +88,80 @@ public static int ErrorQuery (int width, int height, ustring title, ustring mess
...
+		/// <remarks>
+		/// The message box will be vertically and horizontally centered in the container and the size will be automatically determined
+		/// from the size of the title, message. and buttons.
+		/// </remarks>
+		public static int ErrorQuery (ustring title, ustring message, int defaultButton = 0, params ustring [] buttons)
+		{
+			return QueryFull (true, 0, 0, title, message, defaultButton, buttons);
+		}
+
+		static int QueryFull (bool useErrorColors, int width, int height, ustring title, ustring message, int defaultButton = 0, params ustring [] buttons)
```
[the remark this answers is not in the document]
Comment (BDisp) [reply Terminal.Gui/Windows/MessageBox.cs:164]: I ensure it isn't a breaking, because the original parameters are always used as before.

**Pull request merged**